src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -1623,43 +1623,63 @@ void InstanceKlass::do_local_static_fields(void f(fieldDescriptor*, Handle, TRAP
   }
 }
 
-
-static int compare_fields_by_offset(int* a, int* b) {
-  return a[0] - b[0];
-}
-
 void InstanceKlass::do_nonstatic_fields(FieldClosure* cl) {
   InstanceKlass* super = superklass();
   if (super != NULL) {
     super->do_nonstatic_fields(cl);
   }
   fieldDescriptor fd;
   int length = java_fields_count();
-  // In DebugInfo nonstatic fields are sorted by offset.
-  int* fields_sorted = NEW_C_HEAP_ARRAY(int, 2*(length+1), mtClass);
-  int j = 0;
   for (int i = 0; i < length; i += 1) {
     fd.reinitialize(this, i);
     if (!fd.is_static()) {
-      fields_sorted[j + 0] = fd.offset();
-      fields_sorted[j + 1] = i;
-      j += 2;
+      cl->do_field(&fd);
     }
   }
-  if (j > 0) {
-    length = j;
+}
+
+struct F {
```

Comment:
Hi Coleen, is it possible to replace this with existing Pair structure(utilities/pair.hpp)?

Reply:
Yeah, that would be nicer.  Let me have a look.

Reply:
Thank you for the improvement.  I was trying to be minimal with 'F' but Pair is better.